tests.py

```diff
@@ -520,6 +520,30 @@ def test_multi_ifo_fails(self):
                 **self.signal_parameters,
             )
 
+    def test_makefakedata_usecached(self):
+
+        # Make everything from scratch
+        writer = self.writer_class_to_test(
+            outdir=self.outdir,
+            **default_Writer_params,
+            **default_signal_params,
+            **default_transient_params,
+        )
+        writer.make_data(verbose=True)
+        first_time = os.path.getmtime(writer.sftfilepath)
+
+        # Re-run, and should be unchanged
+        writer.make_data(verbose=True)
+        second_time = os.path.getmtime(writer.sftfilepath)
+        self.assertTrue(first_time == second_time)
+
+        # third run: touch the .cff to force regeneration
+        time.sleep(1)  # make sure timestamp is actually different!
+        os.system("touch {}".format(writer.config_file_name))
+        writer.run_makefakedata()
+        third_time = os.path.getmtime(writer.sftfilepath)
+        self.assertFalse(first_time == third_time)
```

Comment:
Ok, so just keep overriding the method and not worry about any duplication, and it should work?

Reply:
Currently working on it. I'll override the method and hack a final check after it. Not beautiful, but will test properly.

Reply:
Ah sorry, I thought we were only talking about _test_ class methods. But it's about the cff logic of the main writer class.

Looks to me like the clean solution to make the main `Writer`'s behaviour more adaptable by children would be:
* in `_basic_setup` only store `self.config_file_name` if `injectionSources` is set
* change logic in `check_cached_data_okay_to_use` correspondingly

Reply:
Your solution is equivalent, because `make_cff` is only called, and hence `os.path.isfile(self.config_file_name)` only becomes true, if we actually do an injection. :+1:

Reply:
Done differently, everyone agrees.